### ISSUE TITLE
No warning for null-conditional operator on non-nullable

### ISSUE THREAD

**Issue by dtivel** [dotnet/roslyn-analyzers#6242]
**Version Used**: 
```
Microsoft Visual Studio Enterprise 2019 Int Preview
Version 16.8.0 Preview 1.0 [30322.17.master]
VisualStudio.16.IntPreview/16.8.0-pre.1.0+30322.17.master
Microsoft .NET Framework
Version 4.8.04084

C# Tools   3.8.0-1.20371.14+ad5096d80a47fb7aa5dbce164fa042804f6a7422
C# components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.
```
**Steps to Reproduce**:

1.  Compile the following code:
```C#
#nullable enable

using System;
using System.IO;

class A : IDisposable
{
    private readonly Stream _stream;

    internal A(Stream stream)
    {
        _stream = stream;
    }

    public void Dispose()
    {
        _stream?.Dispose();
    }
}
```

**Expected Behavior**:  A warning should be issued for the null-conditional operator.  The null-conditional operator conveys intent to guard against the value being `null`, which is misleading given that the value should never be `null`.

**Actual Behavior**:  No warning.

**Comment by mavasani** [comment]
Agree with @CyrusNajmabadi. Warning here can cause a runtime regression. For example, consider the below code in the same project which hasn't enabled nullable:

```csharp
#nullable disable

using System;
using System.IO;

public class PublicType
{
    public void PublicMethod(Stream stream)
    {
        using var a = new A(stream);
    }
}
```
 
Assuming `PublicMethod` is a public API that any third-party library can invoke with potentially null `stream` argument, removing the `?.` operator in your suggested example would lead to a null dereference exception.

In general, nullable enable feature is to aid users to harden their code base against null dereference exceptions, but in presence of mix of nullable enabled and disabled code, there are no guarantees about null-ness of values being passed around. Compilers and analyzers have to be conservative and cannot assume that any variable in nullable enabled code with a `?` is guaranteed to be non-null.
